libraries/wallet/wallet.cpp

```diff
@@ -3186,21 +3190,56 @@ signed_transaction wallet_api::htlc_create( string source, string destination, s
          claim_period_seconds, broadcast);
 }
 
-variant wallet_api::get_htlc(std::string htlc_id) const
+fc::optional<fc::variant> wallet_api::get_htlc(std::string htlc_id) const
 {
-   static detail::htlc_hash_to_string_visitor vtor;
-
-   graphene::chain::htlc_object obj = my->get_htlc(htlc_id);
-   fc::mutable_variant_object ret_val;
-   ret_val["database_id"] = (std::string)obj.id;
-   ret_val["from"] = (std::string)((graphene::db::object_id_type)obj.from);
-   ret_val["to"] = (std::string)((graphene::db::object_id_type)obj.to);
-   ret_val["amount"] = obj.amount.amount.value;
-   ret_val["asset"] = (std::string)((graphene::db::object_id_type)obj.amount.asset_id);
-   ret_val["expiration"] = fc::get_approximate_relative_time_string(obj.expiration);
-   ret_val["preimage_hash"] = obj.preimage_hash.visit( vtor );
-   ret_val["preimage_size"] = obj.preimage_size;
-   return ret_val;
+   fc::optional<htlc_object> optional_obj = my->get_htlc(htlc_id);
+   if (optional_obj)
```

Comment:
I am not sure if this will work, as far as i know `object.valid()` is used to check optionals in bitshares codebase. If we change to `valid()` then `if (!obj.is_null())` in line 687 can be changed for consistency as well.

Reply:
I fixed this location to use the more expressive .valid(). The object on line 687 is not an optional, but a variant. So I left it as .is_null()